File: 1092-maximum-difference-between-node-and-ancestor/1092-maximum-difference-between-node-and-ancestor.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int ma = 0 ;
    void diff(TreeNode * root , int maxi , int mini){
        if (!root) return ;
        ma = max(max(ma , abs(maxi - root->val)) , abs(mini - root->val));
        diff(root->left , max(maxi , root->val) , min(mini , root->val));
        diff(root->right , max(maxi , root->val) , min(mini , root->val));
        return ; 
    }
    int maxAncestorDiff(TreeNode* root) {
        diff(root->left , root->val , root->val);
        diff(root->right , root->val , root->val);
        return ma;
    }   
};

File: 1092-maximum-difference-between-node-and-ancestor/1092-maximum-difference-between-node-and-ancestor.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.ma = 0
    def maxAncestorDiff(self, root: Optional[TreeNode]) -> int:
        def finddiff(root, maxi , mini):
            if not root: return 
 
            self.ma = max(self.ma , abs(maxi - root.val) , abs(mini - root.val))    
            finddiff(root.left , max(maxi , root.val) , min(mini , root.val))
            finddiff(root.right ,max(maxi , root.val) , min(mini , root.val))

        finddiff(root.left,root.val , root.val )
        finddiff(root.right,root.val , root.val)
        return self.ma